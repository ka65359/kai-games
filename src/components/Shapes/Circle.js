////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////
/**
 * @author Kai
 * @version 1.0.0
 * @exports Circle
 *
 * @module Circle
 * @description A module that builds a circle represented on a game board that can be styled
 *  and have a callback function when clicked.
 * @typedef {Object} Circle
 * @property {string} content - The inner contents of circle
 * @property {function} onClick - The function that fires when a circle is clicked
 * @property {'top-left'|'top-right'|'center'|'bottom-left'|'bottom-right'} position - The position
 *  within a grid square of the game board. Currently only the corner and center positions are
 *  supported (default is 'center')
 * @property {'small'|'medium'|'large'|'max'} size - Size of the circle to render
 * @property {string} textColor - The color of the inner content. Takes a CSS.color string or hex value
 * @property {string} bgColor - The background color of the square. Takes a CSS.color string or hex value
 */
////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////
import React from "react";
import PropTypes from "prop-types";
import { color, text } from "@storybook/addon-knobs";
import "./_circle.scss";

const Circle = ({ content, onClick, position, size, textColor, bgColor }) => {
  if (typeof onClick !== "function") {
    onClick = () => {};
  }
  if (!content) {
    content = "";
  }

  let className = "kai-circle circle";
  if (position) {
    className += " kai-shape-pos-" + position;
  }
  if (size) {
    className += " kai-shape-size-" + size;
  }

  let style = {};
  if (color("content color", textColor)) {
    style.color = textColor;
  }
  if (color("background color", bgColor)) {
    style.backgroundColor = bgColor;
  }

  return (
    <div className={className} onClick={onClick} style={style}>
      {text("content", content)}
    </div>
  );
};

Circle.propTypes = {
  content: PropTypes.string,
  onClick: PropTypes.func,
  position: PropTypes.string,
  size: PropTypes.string,
  textColor: PropTypes.string,
  bgColor: PropTypes.string
};

export default Circle;
