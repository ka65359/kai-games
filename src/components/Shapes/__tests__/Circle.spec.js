import React from "react";
import { shallow } from "enzyme";
import Circle from "../Circle";

describe("Circle() tests", () => {
  let data;

  beforeEach(function() {
    data = {
      content: "Test content",
      size: "small",
      position: "top-left",
      bgColor: "blue",
      textColor: "#000",
      onClick: jest.fn()
    };
  });

  afterEach(function() {
    data = null;
  });

  it("should render correctly in mode", () => {
    const component = shallow(<Circle {...data} />);

    expect(component).toMatchSnapshot();
  });
});
