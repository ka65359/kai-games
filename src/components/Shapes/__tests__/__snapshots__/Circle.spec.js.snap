// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle() tests should render correctly in mode 1`] = `ShallowWrapper {}`;
